workflows/downloadraw-workflow.yaml

```diff
@@ -18,58 +23,81 @@ spec:
-          { "activityID": "ScenarioMIP", "experimentID": "ssp126", "tableID": "day", "variableID": "tasmax", "sourceID": "GFDL-ESM4", "institutionID": "NOAA-GFDL", "memberID": "r1i1p1f1", "grid_label": "gr1"},
-          { "activityID": "ScenarioMIP", "experimentID": "ssp126", "tableID": "day", "variableID": "pr", "sourceID": "GFDL-ESM4", "institutionID": "NOAA-GFDL", "memberID": "r1i1p1f1", "grid_label": "gr1"},
-          { "activityID": "CMIP", "experimentID": "historical", "tableID": "day", "variableID": "tasmin", "sourceID": "GFDL-ESM4", "institutionID": "NOAA-GFDL", "memberID": "r1i1p1f1", "grid_label": "gr1"},
-          { "activityID": "CMIP", "experimentID": "historical", "tableID": "day", "variableID": "tasmax", "sourceID": "GFDL-ESM4", "institutionID": "NOAA-GFDL", "memberID": "r1i1p1f1", "grid_label": "gr1"},
-          { "activityID": "CMIP", "experimentID": "historical", "tableID": "day", "variableID": "pr", "sourceID": "GFDL-ESM4", "institutionID": "NOAA-GFDL", "memberID": "r1i1p1f1", "grid_label": "gr1"}
-        ]
 
-  templates:
-  - name: loop-gcmruns
+  - name: main
     inputs:
       parameters:
-      - name: gcmruns-list
+      - name: gcmruns-url
     steps:
-    - - name: download-run
-        template: download
+    - - name: fetch-gcmruns-list
+        template: fetch-gcmruns-list
+        arguments:
+          parameters:
+          - name: url
+            value: "{{ inputs.parameters.gcmruns-url }}"
+    - - name: download-gcm
+        template: download-gcm
         arguments:
           parameters:
           - name: activity-id
-            value: "{{item.activityID}}"
+            value: "{{item.activity_id}}"
           - name: experiment-id
-            value: "{{item.experimentID}}"
+            value: "{{item.experiment_id}}"
           - name: table-id
-            value: "{{item.tableID}}"
+            value: "{{item.table_id}}"
           - name: variable-id
-            value: "{{item.variableID}}"
+            value: "{{item.variable_id}}"
           - name: source-id
-            value: "{{item.sourceID}}"
+            value: "{{item.source_id}}"
           - name: institution-id
-            value: "{{item.institutionID}}"
+            value: "{{item.institution_id}}"
           - name: member-id
-            value: "{{item.memberID}}"
+            value: "{{item.member_id}}"
           - name: grid-label
             value: "{{item.grid_label}}"
+          - name: version
+            value: "{{item.version}}"
           - name: outpath
-            value: "raw/{{item.sourceID}}/{{item.experimentID}}/{{item.memberID}}/{{item.variableID}}/{{item.grid_label}}.zarr"
-        withParam: "{{inputs.parameters.gcmruns-list}}"
+            value: "raw/{{item.source_id}}/{{item.experiment_id}}/{{item.member_id}}/{{item.variable_id}}/{{item.grid_label}}/{{item.version}}.zarr"
```

Comment:
Really glad you added version information - this is a very much needed addition since we may at some point have multiple versions for GCMs. At some point we need to deal with possible errata and newer versions for that reason but this is fine for now. Something to keep in mind though.

Reply:
Okay! Sounds good.